File: Data Structures & Algorithms/add-binary/submission-0.cpp
class Solution {
public:
    string addBinary(string a, string b) {

        int carry = 0;
        string ans = "";

        int i = a.size()-1;
        int j = b.size()-1;
        while(i >= 0 && j >= 0){
            int x = a[i--]-'0';
            int y = b[j--]-'0';

            int res = x+y+carry;
            carry = res/2;
            ans += '0'+(res%2);
        }

        while(i >= 0){
            int x = a[i--]-'0';
            int res = x+carry;
            carry = res/2;
            ans += '0'+(res%2);
        }

        while(j >= 0){
            int x = b[j--]-'0';
            int res = x+carry;
            carry = res/2;
            ans += '0'+(res%2);
        }

        if(carry) ans +='0'+carry;
        reverse(ans.begin(),ans.end());
        return ans;

        
    }
};

File: Data Structures & Algorithms/non-cyclical-number/submission-0.py
class Solution:
    def isHappy(self, n: int) -> bool:

        visited = set()
        visited.add(n)

        def sumOfSquares(n):
            res = 0

            while n > 0:
                res += (n%10)**2
                n //= 10
            return res
        
        while n != 1:
            n = sumOfSquares(n)

            if n in visited:
                return False
            visited.add(n)
        return True
        